### ISSUE TITLE
[Intern Progress] Best Nyah - Outreachy Contibution

### ISSUE THREAD

**Issue by Bee0933** [ersilia-os/outreachy-contributions#9]
# 🌟 Intern Progress Tracker  
Welcome! Use this issue to track your progress as you work through the AI/ML modeling exercise. Follow each step, check the boxes when completed, and update your issue as you go.

# ▶️ **Entry tasks**
- [x] Motivation letter posted in this issue
- [x] Installed the Ersilia Model Hub and successfully run models!
- [x] Recorded system settigns in this issue
- [x] Explained completed tests of Ersilia in the issue

---

##  **1. Download a Dataset of Interest**
- [x] Created a fork of this repository to work on
- [x] Selected a dataset from Therapeutics Data Commons (TDC)
- [x] Verified that the dataset is a classification problem*
- [x] Read and understood the dataset’s background and endpoint  
- [x] Assessed computational feasibility (size, features, etc.)  
- [x] Downloaded the dataset using Python  
- [x] Saved data in the `/data` folder  
- [x] Started writing documentation in the README (Installation, dataset details, how to run code)  

---

##  **2. Featurise the Data**
- [x] Browsed Ersilia Model Hub for "Representation" models  
- [x] Selected an appropriate featuriser and justified my choice  
- [x] Implemented featurisation in a notebook or script
- [x] Ensured the process is reproducible 
- [x] Updated the README with featurisation instructions  

---

##  **3. Build an ML Model**
- [ ] Selected an ML framework
- [ ] Trained the model on featurised data  
- [ ] Evaluated the model (accuracy, precision, recall, etc.)  
- [ ] Created graphs using Matplotlib for better visualization  
- [ ] Ensured all steps are reproducible  
- [ ] Documented the model-building process in README

---

##  **4. Prepare Code for Review**
- [ ] Double-checked that documentation is clear and complete  
- [ ] Evaluated if my model performance is satisfactory 
- [ ] Identified possible improvements  
- [ ] Updated this issue to request a review 

---

##  **5. Stretch Tasks (Optional)**
- [ ] Tried a different **featuriser**  
- [ ] Tested other **ML architectures**  
- [ ] Applied the model to **public datasets (e.g., ChEMBL)**  
- [ ] Any other improvements? (Describe below)  

---

## 📌 **Requesting Review**
Once you’ve completed **Steps 1-4**, request a review by mentioning a mentor:  
👤 *@GemmaTuron* Please review my work!  

---

### 📝 **Asking questions**
Use this issue to update your progress, explain where you are stuck or ask for review!

Thank you for contributing! 🎉

**Comment by Bee0933** [comment]
**Motivation statement**

Hi @GemmaTuron,

I am quite excited about the Outreachy Internship opportunity and the chance to contribute to Ersilia, as it aligns with my aspiration for deepening my knowledge in AI/ML by utilizing my Data Engineering and Infrastructure experience to support positive social impact. My work over time, has been quite centred on personal and some commercial projects, so I'm looking for opportunities to get back to my roots with ML and AI in practice specifically centered on contributing to open-source.

Notably, Ersilia is offering an opportunity to work with AI/ML, which would not only allow me to gain hands-on experience, but also contribute to meaningful projects in healthcare to eradicate infectious diseases. This is a clear application of AI to a critical domain which inspires and motivates me to contribute (although i have little domain knowledge in healthcare 😁), as I believe this can bring alot of positive change especially in resource limited countries.

I believe I would be a valuable asset to Ersilia’s work as my skills closely alligns with  Ersilia’s requirments. I've got some practical experience with Python, Docker, AWS, and developer tools like Git, GitHub Actions, and Pytest. On top of that I have dabbled with ML frameworks such as Scikit-Learn and TensorFlow, and of course not to let out documentation and research which is crucial for this domain. _(i’m also really into visually representing stuff 😎)_ 

With that being said, I beleive this Internship presents me a huge opportunity to grow in my career while also becoming a valuable open-source contributor.  I hope to contribute to the mission of Ersilia in a meaningful way and Post-internship,  continue working with AI, Data, and Infrastructure, ideally in roles that tackle issues like healthcare and sustainability.

I am eager to collaborate with the Ersilia team, learn from experienced mentors, and contribute to the project’s success!

cheers! 🥂

**Comment by Bee0933** [comment]
**System Settings:**

- Operating System: macOS
- Ersilia Version: `ersilia, version 0.1.42`
- Docker Version: `28.0.1`
- Python Version: `conda (Python 3.10.10)`

**Comment by Bee0933** [comment]
**Running the Models**

Following the contribution guidelines outlined in the [Ersilia Book](https://ersilia.gitbook.io/ersilia-book/contributors/internships/outreachy-summer-2025) Documentation, I completed the initial tutorial task of installing the [pre-requisites](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/installation#id-2.1.-pre-requisites) which included:
- Setup up conda package manager and python 
    _ofcourse I went for miniconda becuase who dosn't like lightweight tools?_ 😅

- Setup Git Large File Storage (Git-LFS)  
    _(basically required for many pre-trained models that contain many parameters)_

 I ran **Ersilia Model Hub** on my local machine and executed the specified models with its CLI, which include:

- **[eos3b5e](https://github.com/ersilia-os/eos3b5e)** (calculates molecular weight)
- **[eos2r5a](https://github.com/ersilia-os/eos2r5a)** (calculates Retrosynthetic accessibility score)
- **[eos9ei3](https://github.com/ersilia-os/eos9ei3)** (estimates Synthetic accessibility score)

The models are containerized and packaged then hosted on Ersilia's Docker Hub registry or S3 buckets or GitHub. I prefered to manually pull from the Docker Hub registry to my machine before testing with Ersilia's CLI using: 
`docker pull ersiliaos/eos3b5e:latest`

I successfully tested them using the tasks input SMILES's representation for Ibuprofen: 

`CC(C)CC1=CC=C(C=C1)C(C)C(=O)O.`

- served the fetched **molecular weight** model:
           `ersilia serve eos3b5e` _which Defaults to local models over the cloud precalculation store._
            `ersilla serve --help` _provides other paramaters adjust your results_

- ran the served **molecular weight** model with the Ibuprofen SMILE input:
           `ersilia run -i  "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"` 

_I repeated the process for the other models!!_

<br>

**Output Results:**
The `ersilia serve` command serves the model by running the existing docker image of the specified model on the docker container runtime and assigned a specific port
  
<img width="828" alt="Image" src="https://github.com/user-attachments/assets/8831eadb-42fc-40a0-b74f-d384d1fe29b8" />

<br>
<br>

After serving the model Ersilia allows one to list a catalog of the existing models with `ersilia catalog` which shows all the existing models in your local machine

<img width="669" alt="Image" src="https://github.com/user-attachments/assets/1867bade-c5f3-4dcd-b170-41fe5695bfb0" />

 <br>
<br>

**Running the served models**

- eos3b5e (Molecular weight model):
    
The `ersilia info` command displays model metadata, including a brief description, source code and container registry links and build details. 

<img width="1434" alt="Image" src="https://github.com/user-attachments/assets/9136844d-f119-4a1c-93b4-86bb08abaa49" />

<br>
<br>

- eos2r5a (Retrosynthetic accessibility score model):

<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/6221a557-e2b8-4e6b-8084-247a6f6a07b2" />

<br>
<br>

- eos9ei3 (Synthetic accessibility score model):
    _trying another output format_

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/21f68731-051a-4cbe-a765-8c17b78c695c" />

<img width="649" alt="Image" src="https://github.com/user-attachments/assets/c7e8d657-2b51-4c5c-91c0-80ecfb51689c" />

<br>
<br>

After using the model for various tasks, you can stop the currently served model with `ersilia close` and could choose to remove the model from your local machine if no longer needed using `ersilia delete eos9ei3` 

<br>

`ersilia --help` shows a bunch of other funtionality the CLI provides for interaction wth the Model Hub like:
- `ersilia test` wich allows one to Test a local models that are under development as well as on deployment and
  obtain a detailed report on its expected behavior and performance
- `ersilia auth`  logs in to ersilia to enter contributor mode
- `ersilia example`  generates input examples for the model of interest
- `ersilia uninstall`  uninstalls ersilia _(which is a bad idea now)_ 🌚

<br>
<br>

**Observations:**
- Model Execution: The model runs successfully and returns predictions.

    The model takes in a given chemical structure (in SMILES format), It processes the information and returnes a numerical 
     prediction example in the case of model for  molecular weight which gives the result in **g/mol** which the unit can be 
     confirmed in the `ersilia info` command.

     also an InChIKey is generated from the chemical structure input which is a standard way of representing a chemical 
     substance, like a fingerprint of the molecule.

     ![Image](https://github.com/user-attachments/assets/feae8f18-3a91-48aa-b753-c3bcceecdba3)


- Output Formats: `.json` is the default format; other formats include `.tsv, .csv or .h5`.
- Docker Dependence: The model runs inside a Docker container and requires it to be active.
- Multiple Models: The Ersilia CLI allows one to run a variety of models present on Ersilia's repositories for different tasks.

**Comment by GemmaTuron** [comment]
Hi @Bee0933 

Thanks for this good start to the contribution period, nice schema to summarise the work, I can definitely see your background in data engineering and infrastructure there!
Please go ahead and register your first contribution for Ersilia in the Outreachy page linking this issue, where we will continue working during the rest of the internship. If you can, join in the welcome call today at 17:00 CET for more info!

**Comment by Bee0933** [comment]
Alright!!

Thanks @GemmaTuron for the follow-up, I'd proceed to the next tasks

**Comment by GemmaTuron** [comment]
Hi @Bee0933 

Could you comment here briefly on why you have chosen the HIV dataset and which featuriser?

**Comment by Bee0933** [comment]
sure @GemmaTuron !!

My reason for choosing the HIV dataset is because HIV is a chronic global health problem for which there is curreently no cure, and new drug candidates are urgently required to target drug-resistant strains of HIV. This would be a great opportunity to contribute to help combating the diesease. also

Traditional HTS of HIV inhibitors is time-consuming and costly, but using ML models can really help to accelerate screening by predicting molecular activity, reducing laboratory testing.



The Featurizer I used is the Ersilia Compound Embedding due to some reasons wich include:
- It provides physicochemical properties and bioactivity data, making it more informative for HTS task
- Also the embeddings leverage a pretrained network trained on the popular FS-Mol and ChEMBL datasets, allowing them to work well with different chemical structures
- And lastly each molecule is encoded into a 1024-dimensional feature vectors, which basically preserves key molecular characteristics while making similarity comparisons easier.

**Comment by GemmaTuron** [comment]
Thanks @Bee0933 

Go ahead and start the modelling task, let's see how it works

**Comment by miquelduranfrigola** [comment]
Auto-closing this Outreachy intern-progress tracker as completed — the contribution round has ended. Reopen if follow-up is needed.
